src/google-classroom-extractor/google_classroom_extractor/api/resource_sync.py

```diff
@@ -0,0 +1,241 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import logging
+from datetime import datetime
+from pandas import DataFrame, read_sql_query
+import sqlalchemy
+import xxhash
+
+logger = logging.getLogger(__name__)
+
+
+def cleanup_after_sync(resource_name: str, sync_db: sqlalchemy.engine.base.Engine):
+    """
+    Delete sync temporary tables if they exist
+
+    Parameters
+    ----------
+    resource_name: str
+        the name of the API resource, e.g. "Courses", to be used in SQL
+    sync_db: sqlalchemy.engine.base.Engine
+        an Engine instance for creating database connections
+    """
+    with sync_db.connect() as con:
+        con.execute(f"DROP TABLE IF EXISTS Sync_{resource_name}")
+        con.execute(f"DROP TABLE IF EXISTS Unmatched_{resource_name}")
+
+
+def sync_to_db_without_cleanup(
```

Comment:
I split out db temp table cleanup for testability. Integration tests can confirm that the intermediate work done in the temp tables is as expected.